tests/functional/z_cluster/nodes/test_nodes_restart_hci.py

```diff
@@ -378,3 +379,33 @@ def test_stop_provider_mgr_node_and_check_client_connection(
         logger.info(f"Starting the MGR node {mgr_node.name}")
         nodes.start_nodes(nodes=[mgr_node], wait=True)
         self.sanity_helpers.health_check_provider_mode(tries=40)
+
+    @tier4a
+    @pytest.mark.parametrize(
+        "cluster_type",
+        [HCI_CLIENT],
+    )
+    def test_hosted_cluster_nodes_restart(self, cluster_type, nodes):
```

Comment:
The current context client cluster can be hosted cluster based on kubevirt or agent and non- hosted cluster. The test case need to identify the hosted cluster among the clients if the purpose is to run  only on hosted clusters.

Or the test case can be "test_client_cluster_nodes_restart". In this we need either 
* parameterized way of using client clusters of different types (kubevirt, agent, non-hosted and other platforms in future) and also skip each parameterized test case based on the unavailability of the targeted client cluster type.
or
*
Individual tests for client clusters based on kubevirt, non-hosted or agent(agent is not implemented).


Same for the test test_hosted_cluster_nodes_restart_by_stop_and_start

Reply:
I will think of a solution.

Reply:
I will implement the first option, meaning I will add another param for picking up only the relevant Managed clusters(Ohad prefers the term "Managed cluster" than "Client cluster") with a specific type.

Reply:
@jilju Is Agent cluster also considered a hosted cluster?

Reply:
Yes